content/strategy-goals/strategy/code-graph/search/product.md

```diff
@@ -10,37 +10,71 @@ Quicklinks:
+> Python function for k-means clustering
 
-#### Details
+We'd want to see the model generate a query like:
 
-**Notebooks** achieve product market fit and enable individual developers and teams to easily document codebases and share best practices across an organization by enhancing Notebooks with sharing and duplication functionality, novel Notebook creation tools, and useful templates. Developers can already embed search results, code snippets and more, in a markdown-based notebook that creates more robust documentation. For example, by using structural search to persistently capture a function and its contents, even if it's moved in the codebase.
+`file:.\py$ kmeans select:symbol.function`
 
-**Code monitor integrations** will provide the ability to integrate Code monitors with any service that supports webhooks, and will enable multiple actions to be triggered by a Code monitor, including Slack notifications. We've also lifted the current fifty repository limit on monitors to 1,000 repositories, soon to be over 10,000.
+Learning Sourcegraph's query language is a struggle for many users, to the point that they simply never go beyond using literal search. Anything we can do to reduce the learning curve and improve result quality will have a big impact on user experience and potentially drive more usage.
 
-**Core platform investments** like Notebooks and Code monitoring integrations are how we plan to position Sourcegraph as the code search platform. We need to express the power of our code search through multiple avenues so it's clear why best-in-class code search is a prerequisite for fostering engineering capability. Extending and integrating the core platform products and features to servve our use cases will help the market see both how essential great code search is, and that it's a _part_ of Sourcegraph's value proposition, not its only value proposition.
+If successful, this effort would have implications beyond Code Search. Imagine being able to create a Code Insight or a Batch Change with natural language.
 
 #### What we're not working on and why
 
-While we have R&D efforts in various areas, we have no implementation plans for machine learning-based ranking or relevance improvements. ML applications in code search remain a challenge. If we were to invest in this area it would likely be closer to semantic searching than a product like GitHub Copilot. While there's a lot of value in high quality semantic code search, there are also many unique applications that can be unlocked by staying focused on–and continuing to extend–our distinctive approach to code search.
+This quarter, we are largely deferring feature work in favor of foundational and research efforts. Q1 was a hugely productive quarter for the team, and we want to let the new products and features that shipped get some usage and garner feedback.
```

Comment:
```suggestion
This quarter, we are largely deferring launching new features in favor of foundational and research efforts. Q1 was a hugely productive quarter for the team, and we want to let the new products and features that shipped get some usage and garner feedback.
```

I think this might convey what you mean better?